### ISSUE TITLE
`ansible_` prefixed variables are flagged erroneously in `set_fact` task by `var-naming`

### ISSUE THREAD

**Issue by BendingBender** [ansible/ansible-lint#3501]
##### Summary
Since at least ansible-lint 6.16.1 the `var-naming` rule flags `ansible_`-prefixed variables when they're defined in a `set_fact` task. As far as I see it, they shouldn't be flagged because there's no way to influence the naming here.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.16.2 using ansible-core:2.14.6 ruamel-yaml:0.17.26 ruamel-yaml-clib:0.2.7
```

- ansible installation method:  homebrew
- ansible-lint installation method: homebrew

##### STEPS TO REPRODUCE

Use a `set_fact` task in a role like the following:

```console (paste below)
- name: Set private key path to use to connect to host to the initial connection key
  ansible.builtin.set_fact:
    ansible_ssh_private_key_file: "{{ role_name_some_path }}"
```

##### Desired Behavior

`ansible-lint` shouldn't flag `ansible_` prefixed variables.

##### Actual Behavior

Please give some details of what is happening.
Include a [minimum complete verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

`my_role/tasks/main.yml`:
```yaml
- name: Set private key path to use to connect to host to the initial connection key
  ansible.builtin.set_fact:
    ansible_ssh_private_key_file: /some/path
```

`example_playbook.yml`:
```yaml
- name: Run example role
  hosts: some_host
  gather_facts: false
  roles:
    - my_role
```

```sh
ansible-lint

var-naming[no-role-prefix]: Variables names from within roles should use role_name_ as a prefix. (set_fact: ansible_ssh_private_key_file)
roles/my_role/tasks/main.yml:3 Task/Handler: Set private key path to use to connect to host to the initial connection key

Read documentation for instructions on how to ignore specific rule violations.

                    Rule Violation Summary                     
 count tag                        profile rule associated tags 
     1 var-naming[no-role-prefix] basic   idiom                

Failed after min profile: 1 failure(s), 0 warning(s) on 2 files.
```
[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by BendingBender** [comment]
Duplicates https://github.com/ansible/ansible-lint/issues/3456.
